Deque Axe Integration Page (#361)
saucelabs/sauce-docs: docs/basics/integrations/deque.md

---
id: deque
title: "Deque axe™ Integration"
sidebar_label: "Axe Deque"
keywords:
- accessibility-testing
- accessibility
- automated-testing
- how-to
---

<p><small><span class="highlight beta">BETA</span></small></p>

[Deque's axe™](https://www.deque.com/axe/) is one of the world's leading digital accessibility toolkits. The Sauce Labs integration allows you to run your accessibility tests, using axe™, on our platform.

## What You'll Need
* [A Sauce Labs Account](https://saucelabs.com/sign-up)
* [A Deque Account (Optional)](https://axe.deque.com/plans)
* [Install Git](https://git-scm.com/downloads)
* [Install Node/NPM](https://nodejs.org/en/download/)

## Project Setup

Clone or download the [example project](https://github.com/saucelabs/axe-wdio):

> ```bash

Reviewer: It probably doesn't matter, but are we assuming that anyone who is using SSL will know what they are doing and won't care that this code sample is HTTPS?

Author: yea it doesn't matter, that's why I put the link to the project and the sample code (copied from github). They should be able to figure it out if they're going to run the test.

git clone https://github.com/saucelabs/axe-wdio.git
```

> From the main project directory, change the following lines in `wdio.saucelabs.conf.js`:

Reviewer: ```suggestion
From the main project directory, edit `wdio.saucelabs.conf.js` to set your Sauce Labs account values for the highlighted config in the following sample:
```

     * Desktop browsers
     */
    {
        browserName: 'googlechrome',
        platformName: 'Windows 10',
        browserVersion: 'latest',
        'sauce:options': {
            ...defaultBrowserSauceOptions,
        },
    },
    {
        browserName: 'firefox',
        platformName: 'Windows 10',
        browserVersion: 'latest',
        'sauce:options': {
            ...defaultBrowserSauceOptions,
        },
    },
    // {
    //     browserName: 'internet explorer',
    //     platformName: 'Windows 8.1',
    //     browserVersion: 'latest',
    //     'sauce:options': {
    //         ...defaultBrowserSauceOptions,
    //     },
    // },
    {
        browserName: 'MicrosoftEdge',
        platformName: 'Windows 10',
        browserVersion: 'latest',
        'sauce:options': {
            ...defaultBrowserSauceOptions,
        },
    },
    // Safari 13
    {
        browserName: 'safari',
        platformName: 'macOS 10.15',
        browserVersion: 'latest',
        'sauce:options': {
            ...defaultBrowserSauceOptions,
        },
    },
];

// ========
// Services
// ========
config.services = config.services.concat('sauce');

exports.config = config;

```

> ### Sauce Labs Regions

Reviewer: This heading is too restrictive, since the warning below applies to all three configs, not just region. Maybe something like `###  Sauce Labs Configs`?

Author: 👍

> ### Sauce Labs Regions

Author: ```suggestion
### Sauce Labs Configs
```

Specify the Sauce Labs data center location in which you want to run your tests with the `region` property. 
   

> Available short handles for regions are:

Author: ```suggestion
The available data center locations are: United States West and European Central. 

The short-hand values that this script accepts are:
```

>    * `us` - *default*

Reviewer: 1. Probably obvious, but should the short handles include the spelled out region?
2. If this setting has a "default value", why does the next sentence say it is a required property?

Author: 1. I don't know, it was worse before so we can put the full values in () or something
2. True a little redundant, but it more so means it's required property for the **script** to run the way Deque set it up.

Author: @sweeneyskirt-sl added a suggestion ☝️

   * `eu`

This `region` property is required, and used for both Virtual Machine Cloud and Real Device Cloud tests.

:::warning Don't Hard-Code Credentials
Hard-coding your `user`, `key`, and `region` property values is not a recommended best practice due to security reasons. 

Instead, export your [Sauce Labs Account Credentials as Environment Variables](https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Use+Environment+Variables+for+Authentication+Credentials) and add them in your script like the following example:

```js
config.user = process.env.SAUCE_USERNAME;
config.key = process.env.SAUCE_ACCESS_KEY;
config.region = process.env.REGION || 'us';
```

Visit the [WebDriverIO documentation](https://webdriver.io/docs/sauce-service.html) for further details.
:::

## Run the Test
Save the file and run the following command from the main project directory:
```bash
npm install && npm test
```

> 

Reviewer: What happens when you run the test? What kind of results do you get?

Author: I don't know, and I can't run it until I get access to the experiment. I don't want to paste the output because it might change soon so I decided to keep this one sparse. #underpromise
